contracts/src/v0.8/automation/chains/ScrollModule.sol

```diff
@@ -15,14 +19,22 @@ contract ScrollModule is ChainModuleBase {
   address private constant SCROLL_ORACLE_ADDR = 0x5300000000000000000000000000000000000002;
   IScrollL1GasPriceOracle private constant SCROLL_ORACLE = IScrollL1GasPriceOracle(SCROLL_ORACLE_ADDR);
 
+  /// @dev L1 fee coefficient can be applied to reduce possibly inflated gas cost
+  uint8 public s_l1FeeCoefficient = 100;
   uint256 private constant FIXED_GAS_OVERHEAD = 45_000;
   uint256 private constant PER_CALLDATA_BYTE_GAS_OVERHEAD = 170;
 
-  function getCurrentL1Fee() external view override returns (uint256) {
-    return SCROLL_ORACLE.getL1Fee(bytes.concat(msg.data, SCROLL_L1_FEE_DATA_PADDING));
+  constructor() ConfirmedOwner(msg.sender) {}
+
+  function getCurrentL1Fee(uint256 dataSize) external view override returns (uint256) {
+    return (s_l1FeeCoefficient * _getL1Fee(dataSize)) / 100;
   }
 
   function getMaxL1Fee(uint256 dataSize) external view override returns (uint256) {
+    return _getL1Fee(dataSize);
+  }
+
+  function _getL1Fee(uint256 dataSize) internal view returns (uint256) {
```

Comment:
In theory, you could save quite an amount of gas by implementing the calculation by yourself according to these docs: https://docs.scroll.io/en/developers/transaction-fees-on-scroll/#calculating-the-l1-data-fee-with-gas-oracle

According to the docs, the formula is:
```
l1Gas = zeros * TX_DATA_ZERO_GAS + (nonzeros + 4) * TX_DATA_NON_ZERO_GAS
l1GasFee = ((l1Gas + overhead) * l1BaseFee * scalar) / PRECISION
```

In your case (since you are assuming the worst case with all non-zero bytes):
```
uint256 l1Gas = (dataSize + 4) * 16;
return ((l1Gas + SCROLL_ORACLE.overhead()) * SCROLL_ORACLE.l1BaseFee() * SCROLL_ORACLE.scalar()) / 10**9;
```

Reply:
discussed briefly offchain. we agreed that this improvement can be implemented in the next version.
in order to keep the change size as small as possible, we will use the provided function directly.